astropy/wcs/wcsapi/low_level_api.py

```diff
@@ -159,7 +165,7 @@ def world_axis_object_components(self):
 
     @property
     @abc.abstractmethod
-    def world_axis_object_classes(self):
+    def world_axis_object_classes(self) -> Mapping[str, Tuple[Union[str, type], tuple, dict, Optional[Callable]]]:
```

Comment:
I am not 100% sure this `Optional` is actually specifying what I want here, according to [the docs](https://docs.python.org/3/library/typing.html#typing.Optional) it's the same as `Union[X, None]` which I think implies this tuple must always be 4 long.

Reply:
Does the validation fail if the tuple has three elements?

If so then what about ``Union[Mapping[str, Tuple[Union[str, type], tuple, dict]], Mapping[str, Tuple[Union[str, type], tuple, dict, Optional[Callable]]]]``